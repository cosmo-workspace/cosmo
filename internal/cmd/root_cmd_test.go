package cmd

import (
	"bytes"
	"testing"

	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
)

func TestCommandRoot(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl root suite")
}

var _ = Describe("help", func() {
	It("should match snapshot", func() {
		o := cli.NewRootOptions()
		rootCmd := NewRootCmd(o)

		out := bytes.Buffer{}
		rootCmd.SetOut(&out)
		rootCmd.SetArgs([]string{"--help"})
		err := rootCmd.Execute()
		Expect(err).ShouldNot(HaveOccurred())
		Expect(out.String()).To(MatchSnapShot())
	})
})
