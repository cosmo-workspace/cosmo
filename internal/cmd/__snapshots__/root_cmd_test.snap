['help should match snapshot 1']
SnapShot = """

Command line tool for cosmo API
Complete documentation is available at http://github.com/cosmo-workspace/cosmo

MIT 2024 cosmo-workspace/cosmo

Usage:
  cosmoctl [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  create      Create cosmo resources
  delete      Delete cosmo resources
  get         Get cosmo resources
  help        Help about any command
  login       Login to COSMO Dashboard Server
  resume      Start stopped workspaces
  suspend     Suspend workspaces
  template    Manipulate Template resource
  user        Manipulate User resource
  version     Print the version number
  workspace   Manipulate Workspace resource

Flags:
      --config string          cosmoctl config file path. env:COSMOCTL_CONFIG (default: $HOME/.config/cosmocfg)
      --context string         kube-context (default: current context)
      --dashboard-url string   COSMO Dashboard server endpoint URL. env:COSMOCTL_DASHBOARD_URL
  -h, --help                   help for cosmoctl
  -k, --kube                   use kubernetes API client instead of cosmo dashboard API client
      --kubeconfig string      kubeconfig file path. env:KUBECONFIG (default: $HOME/.kube/config)
  -v, --verbose int            log level. -1:DISABLED, 0:INFO, 1:DEBUG, 2:ALL

Use \"cosmoctl [command] --help\" for more information about a command.
"""
